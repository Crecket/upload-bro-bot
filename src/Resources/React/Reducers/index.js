import {combineReducers} from "redux";

import modal from "./modalReducers";
import user from "./user";

export default combineReducers({
    modal,
    user,
})
