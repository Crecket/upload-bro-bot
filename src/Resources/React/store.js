import {applyMiddleware, createStore} from "redux";
import thunk from "redux-thunk";
import promise from "redux-promise-middleware";
import logger from "redux-logger";

// fetch all reducers as a bundle
import reducer from "./Reducers/index.js";

// create middleware with logger
var middleware = applyMiddleware(promise(), thunk)
if (process.env.DEBUG === "true") {
    // create middleware with logger
    middleware = applyMiddleware(promise(), thunk, logger())
}


//return the store
export default createStore(reducer, middleware)
