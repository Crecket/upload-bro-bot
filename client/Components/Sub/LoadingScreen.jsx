"use strict";
import React from "react";

// get loader types and return setup
const getLoaderType = (type) => {
    switch (type) {
        case "cube-grid": {
            return (
                <div className="sk-cube-grid">
                    <div className="sk-cube sk-cube1"></div>
                    <div className="sk-cube sk-cube2"></div>
                    <div className="sk-cube sk-cube3"></div>
                    <div className="sk-cube sk-cube4"></div>
                    <div className="sk-cube sk-cube5"></div>
                    <div className="sk-cube sk-cube6"></div>
                    <div className="sk-cube sk-cube7"></div>
                    <div className="sk-cube sk-cube8"></div>
                    <div className="sk-cube sk-cube9"></div>
                </div>
            );
        }
        case "rotating-plane": {
            return (
                <div class="rotating-plane"></div>
            );
        }
        case "double-bounce": {
            return (
                <div>
                    <div class="double-bounce1"></div>
                    <div class="double-bounce2"></div>
                </div>
            );
        }
        case "rectangle-bounce": {
            return (
                <li class="rectangle-bounce">
                    <div class="rect1"></div>
                    <div class="rect2"></div>
                    <div class="rect3"></div>
                    <div class="rect4"></div>
                    <div class="rect5"></div>
                </li>
            );
        }
        default: {
            throw new Error('Invalid loading type');
        }
    }
}

class LoadingScreen extends React.PureComponent {
    constructor(props, context) {
        super(props, context);
        this.state = {};
    }

    render() {
        return (
            <div className="sk-loader">
                {getLoaderType(this.props.loadingType || "cube-grid")}
            </div>
        );
    }
}

export default LoadingScreen;
