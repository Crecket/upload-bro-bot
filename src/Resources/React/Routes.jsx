import React from 'react';
import {Route, IndexRoute} from 'react-router';

import Main from './Components/Main';
import Home from './Pages/Home';
import Login from './Pages/Login';
import NotFound from './Pages/NotFound';
import DropboxLoginCallback from './Pages/DropboxLoginCallback';
import GoogleLoginCallback from './Pages/GoogleLoginCallback';

export default (
    <Route path="/" component={Main}>
        {/* default route */}
        <IndexRoute component={Home}/>

        {/* Routes*/}
        <Route path='/login' component={Login}/>
        <Route path='/login/google/callback' component={GoogleLoginCallback}/>
        <Route path='/login/dropbox/callback' component={DropboxLoginCallback}/>

        {/* fall back route, if no others are found show 404*/}
        <Route path='*' component={NotFound}/>
    </Route>
);