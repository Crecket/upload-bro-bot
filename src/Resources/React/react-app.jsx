import React from 'react';
import ReactDOM  from 'react-dom';
import {Provider} from "react-redux";
import injectTapEventPlugin  from 'react-tap-event-plugin';
import Logger from './Helpers/Logger.js';

// main app
import Main from './Components/Main';
import store from "./store";

// injection, required for materialze tap events
injectTapEventPlugin();

// render the react app
ReactDOM.render(
    <Provider store={store}>
        <Main />
    </Provider>,
    document.getElementById('app')
);

Logger.debug('Mounted react succesfully');
