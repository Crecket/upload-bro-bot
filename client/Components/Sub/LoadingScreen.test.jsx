import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

import LoadingScreen from "./LoadingScreen.jsx";

describe("<LoadingScreen />", () => {
    it("matches default snapshot", () => {
        const tree = renderer.create(<LoadingScreen />);
        expect(tree).toMatchSnapshot();
    });

    it("throws an error for giving invalid loading tpye", () => {
        expect(() => {
            "use strict";
            const tree = renderer.create(
                <LoadingScreen loadingType="invalid-type-123" />
            );
        }).toThrow("Invalid loading type");
    });

    it("matches cube-grid snapshot", () => {
        const tree = renderer.create(<LoadingScreen loadingType="cube-grid" />);
        expect(tree).toMatchSnapshot();
    });

    it("matches rotating-plane snapshot", () => {
        const tree = renderer.create(
            <LoadingScreen loadingType="rotating-plane" />
        );
        expect(tree).toMatchSnapshot();
    });

    it("matches double-bounce snapshot", () => {
        const tree = renderer.create(
            <LoadingScreen loadingType="double-bounce" />
        );
        expect(tree).toMatchSnapshot();
    });

    it("matches rectangle-bounce snapshot", () => {
        const tree = renderer.create(
            <LoadingScreen loadingType="rectangle-bounce" />
        );
        expect(tree).toMatchSnapshot();
    });
});
