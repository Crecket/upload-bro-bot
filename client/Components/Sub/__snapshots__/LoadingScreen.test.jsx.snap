// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingScreen /> matches cube-grid snapshot 1`] = `
<div
  className="sk-loader"
>
  <div
    className="sk-cube-grid"
  >
    <div
      className="sk-cube sk-cube1"
    />
    <div
      className="sk-cube sk-cube2"
    />
    <div
      className="sk-cube sk-cube3"
    />
    <div
      className="sk-cube sk-cube4"
    />
    <div
      className="sk-cube sk-cube5"
    />
    <div
      className="sk-cube sk-cube6"
    />
    <div
      className="sk-cube sk-cube7"
    />
    <div
      className="sk-cube sk-cube8"
    />
    <div
      className="sk-cube sk-cube9"
    />
  </div>
</div>
`;

exports[`<LoadingScreen /> matches default snapshot 1`] = `
<div
  className="sk-loader"
>
  <div
    className="sk-cube-grid"
  >
    <div
      className="sk-cube sk-cube1"
    />
    <div
      className="sk-cube sk-cube2"
    />
    <div
      className="sk-cube sk-cube3"
    />
    <div
      className="sk-cube sk-cube4"
    />
    <div
      className="sk-cube sk-cube5"
    />
    <div
      className="sk-cube sk-cube6"
    />
    <div
      className="sk-cube sk-cube7"
    />
    <div
      className="sk-cube sk-cube8"
    />
    <div
      className="sk-cube sk-cube9"
    />
  </div>
</div>
`;

exports[`<LoadingScreen /> matches double-bounce snapshot 1`] = `
<div
  className="sk-loader"
>
  <div>
    <div
      class="double-bounce1"
    />
    <div
      class="double-bounce2"
    />
  </div>
</div>
`;

exports[`<LoadingScreen /> matches rectangle-bounce snapshot 1`] = `
<div
  className="sk-loader"
>
  <li
    class="rectangle-bounce"
  >
    <div
      class="rect1"
    />
    <div
      class="rect2"
    />
    <div
      class="rect3"
    />
    <div
      class="rect4"
    />
    <div
      class="rect5"
    />
  </li>
</div>
`;

exports[`<LoadingScreen /> matches rotating-plane snapshot 1`] = `
<div
  className="sk-loader"
>
  <div
    class="rotating-plane"
  />
</div>
`;
